python/ray/worker.py

```diff
@@ -462,51 +469,52 @@ def get_object(self, object_ids):
-                    [ray.ObjectID(unready_id)], False)
-            # Do another fetch for objects that aren't available locally yet,
-            # in case they were evicted since the last fetch. We divide the
-            # fetch into smaller fetches so as to not block the manager for a
-            # prolonged period of time in a single call.
-            object_ids_to_fetch = list(
-                map(plasma.ObjectID, unready_ids.keys()))
-            ray_object_ids_to_fetch = list(
-                map(ray.ObjectID, unready_ids.keys()))
-            for i in range(0, len(object_ids_to_fetch),
-                           ray._config.worker_fetch_request_size()):
-                if not self.use_raylet:
-                    self.plasma_client.fetch(object_ids_to_fetch[i:(
-                        i + ray._config.worker_fetch_request_size())])
-                else:
-                    self.local_scheduler_client.reconstruct_objects(
-                        ray_object_ids_to_fetch[i:(
-                            i + ray._config.worker_fetch_request_size())],
-                        True)
-            results = self.retrieve_and_deserialize(
-                object_ids_to_fetch,
-                max([
-                    ray._config.get_timeout_milliseconds(),
-                    int(0.01 * len(unready_ids))
-                ]))
-            # Remove any entries for objects we received during this iteration
-            # so we don't retrieve the same object twice.
-            for i, val in enumerate(results):
-                if val is not plasma.ObjectNotAvailable:
-                    object_id = object_ids_to_fetch[i].binary()
-                    index = unready_ids[object_id]
-                    final_results[index] = val
-                    unready_ids.pop(object_id)
-
-        # If there were objects that we weren't able to get locally, let the
-        # local scheduler know that we're now unblocked.
-        if was_blocked:
-            self.local_scheduler_client.notify_unblocked()
+
+        if len(unready_ids) > 0:
+            with self.reconstruction_lock:
+                # Try reconstructing any objects we haven't gotten yet. Try to get them
+                # until at least get_timeout_milliseconds milliseconds passes, then
+                # repeat.
+                while len(unready_ids) > 0:
+                    for unready_id in unready_ids:
+                        self.local_scheduler_client.reconstruct_objects(
+                            [ray.ObjectID(unready_id)], False)
+                    # Do another fetch for objects that aren't available locally yet,
+                    # in case they were evicted since the last fetch. We divide the
+                    # fetch into smaller fetches so as to not block the manager for a
+                    # prolonged period of time in a single call.
+                    object_ids_to_fetch = list(
+                        map(plasma.ObjectID, unready_ids.keys()))
```

Comment:
nit: I understand this is legacy code, but can we take this opportunity to be pythonic here? https://stackoverflow.com/a/10973817/2213289